django/db/models/sql/compiler.py

```diff
@@ -802,6 +809,35 @@ def get_related_klass_infos(klass_info, related_klass_infos):
                 )
         return related_klass_infos
 
+    def get_select_for_update_of_arguments(self):
+        """
+        Return a quoted list of arguments to the SELECT FOR UPDATE OF
+        part of the query.
+        """
+        result = []
+
+        for name in self.query.select_for_update_of:
+            parts = [] if name == 'self' else name.split(LOOKUP_SEP)
```

Comment:
I found

```
django/db/models/fields/related.py:RECURSIVE_RELATIONSHIP_CONSTANT = 'self'
```

I don't think it's appropriate for this case. I can define a new constant if you think it will be useful. Or perhaps we can rename this constant to something more generic like `OWN_MODEL_CONSTANT`.